test/diagnostics/no_warnings_as_errors.swift

```diff
@@ -0,0 +1,9 @@
+// RUN: %empty-directory(%t)
+// RUN: not %target-swift-frontend -typecheck -verify -warnings-as-errors -no-warnings-as-errors %s 2>&1 | %FileCheck %s
```

Comment:
Can we also add a test here to revert the positions of `-warnings-as-errors` and `-no-warnings-as-errors`? to make sure the negation is order-insensitive. Otherwise, LGTM!

Reply:
Wait, do we want the negation to be order-insensitive? What I had in mind is:
`-warnings-as-errors -no-warnings-as-errors` behaves as `-no-warnings-as-errors`
and
`-no-warnings-as-errors -warnings-as-errors` behaves as `-warnings-as-errors`

That way the build system can override either behaviour by appending the opposite option to the build command.

Reply:
@nkcsgexi the order-sensitive behavior is as expected. We do the same for other flags (e.g. `-whole-module-optimization` and `-no-whole-module-optimization`) and Clang also does this, so it shouldn't be order-insensitive here.

Reply:
Ah, sure. I looked into the API you are calling here: 
```  
/// hasFlag - Given an option \p Pos and its negative form \p Neg, return
  /// true if the option is present, false if the negation is present, and
  /// \p Default if neither option is given. If both the option and its
  /// negation are present, the last one wins.
  bool hasFlag(OptSpecifier Pos, OptSpecifier Neg, bool Default=true) const;
```
It seems being order-sensitive is idiomatic, which we also need a test for. And I think you did👍